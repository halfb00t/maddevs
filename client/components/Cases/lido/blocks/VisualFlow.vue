<template>
  <section class="case-lido__visual-flow m-96_bottom media-m-48_bottom">
    <div class="container_regular m-48_bottom media-m-24_bottom">
      <h4 class="case-lido__visual-flow-title m-24_bottom media-m-12_bottom">
        Hereâ€™s visual flow:
      </h4>
      <TextParagraph>
        Also, 100% test coverage was a prerequisite, as well as a test that would pass the entire Smart Contract procedure.
      </TextParagraph>
    </div>
    <div class="container_full background-color-silver p-72_top p-56_bottom m-48_bottom media-p-32_top media-p-32_bottom media-m-24_bottom">
      <div class="container_regular">
        <LottieMad
          class="case_lottie"
          height="603.24px"
          :lottie-link="$getMediaFromS3(`/images/Cases/lido/lottie/visual-flow.json`)"
          :autoplay="true"
        />
      </div>
    </div>
    <div class="container_regular">
      <EmbedCard v-bind="testingProcessData" />
    </div>
  </section>
</template>
<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import LottieMad from '@/components/shared/LottieMad'
import EmbedCard from '@/components/Cases/shared/EmbedCard'
import { testingProcessData } from '@/data/caseLido'

export default {
  name: 'VisualFlow',
  components: { EmbedCard, TextParagraph, LottieMad },
  data() {
    return {
      testingProcessData,
    }
  },
}
</script>
<style scoped lang="scss">
.case-lido__visual-flow {
  &-title {
    @include font('Inter', 28px, 700);
    line-height: 32px;
    letter-spacing: -1px;
    color: $text-color--black-oil;
  }
}
</style>
