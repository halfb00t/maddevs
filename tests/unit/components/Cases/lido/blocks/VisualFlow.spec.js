import { render } from '@testing-library/vue'
import { createLocalVue } from '@vue/test-utils'
import VisualFlow from '../../../../../../client/components/Cases/lido/blocks/VisualFlow'
// eslint-disable-next-line import/order
import lazyLoad from 'nuxt-lazy-load/lib/module'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('Lido VisualFlow component', () => {
  it('should correctly render component', () => {
    const { container } = render(VisualFlow, {
      mocks,
      localVue,
    })

    expect(container).toMatchSnapshot()
  })
})
