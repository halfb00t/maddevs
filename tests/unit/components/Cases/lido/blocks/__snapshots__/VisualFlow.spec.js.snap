// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lido VisualFlow component should correctly render component 1`] = `
<div>
  <section
    class="case-lido__visual-flow m-72_bottom media-m-48_bottom"
  >
    <div
      class="container_regular m-48_bottom media-m-24_bottom"
    >
      <h4
        class="case-lido__visual-flow-title m-24_bottom media-m-12_bottom"
      >
        
      Hereâ€™s visual flow:
    
      </h4>
       
      <p
        class="case_paragraph"
      >
        
      Also, 100% test coverage was a prerequisite, as well as a test that would pass the entire Smart Contract procedure.
    
      </p>
    </div>
     
    <div
      class="container_full background-color-silver p-72_top p-56_bottom m-48_bottom media-p-32_top media-p-32_bottom media-m-24_bottom"
    >
      <div
        class="container_regular"
      >
        <div
          class="lottie-container case_lottie"
          style="--height: 603.24px; --width: 100%;"
        />
      </div>
    </div>
     
    <div
      class="container_regular"
    >
      <div
        class="embed-card"
        data-aos="fade-up"
      >
        <img
          alt="checkbox-circle logo"
          class="embed-card__image"
          data-src="img.jpg"
        />
         
        <div
          class="embed-card__text-content"
        >
          <h4
            class="embed-card__title"
          >
            
      Testing process and how we solved it
    
          </h4>
           
          <a
            class="embed-card__link"
            href="https://github.com/lidofinance/1inch-rewards-manager/blob/main/tests/test_happy_path.py#L13-L34"
            target="_blank"
          >
            
      View on Github
    
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
